File: 06_singleton_pattern/cpp/Singleton.cpp
#include "Singleton.h"

Singleton *Singleton::getInstance(const char *value) {
    if (Singleton::instance == nullptr)
    {
        Singleton::instance = new Singleton(value);
    }
    return Singleton::instance;
}

Singleton::Singleton(const char *value) {
    this->value = value;
}

void Singleton::doStuff() const {
    printf("Value is %s\n", this->value);
}

Singleton* Singleton::instance = nullptr;

File: 06_singleton_pattern/php/singleton.php
<?php

class Singleton {

  private static $instance = null;
  
  private function __construct()
  {
  }

  public static function getInstance()
  {
    if (self::$instance == null)
    {
      self::$instance = new Singleton();
    }
 
    return self::$instance;
  }
}
 